python/runtime/cudaq/algorithms/py_observe.cpp

```diff
@@ -185,75 +196,110 @@ async_observe_result pyObserveAsync(kernel_builder<> &kernel,
 void bindObserve(py::module &mod) {
   auto parallelSubmodule = mod.def_submodule("par");
 
-  py::class_<cudaq::par::mpi>(parallelSubmodule, "mpi");
-  py::class_<cudaq::par::thread>(parallelSubmodule, "thread");
+  py::class_<cudaq::par::mpi>(
+      parallelSubmodule, "mpi",
+      "Type indicating that observe execution should distribute term "
+      "expectation value computation amongst available MPI ranks and GPUs.");
```

Comment:
Will these types ever be reused for sample? If so, it might be worth pulling them out into their own file. There's also this [MPI submodule in _cudaq.cpp](https://github.com/NVIDIA/cuda-quantum/blob/0ad071f24e460c106530c1d09cb0c5d35467cee7/python/cudaq/_cudaq.cpp#L48) that I'm curious how it fits in here? Can we just put these in that submodule instead?

Reply:
Thus far these are only applicable to observe.